packages/core/src/providers/chainState/multiChainStates/provider.tsx

```diff
@@ -27,14 +26,15 @@ interface Props {
 }
 
 function composeChainState(networks: Providers, state: State, multicallAddresses: Props['multicallAddresses']) {
-  return fromEntries(
-    Object.keys(networks).map((chainId) => [
-      Number(chainId),
-      {
+  return Object.keys(networks).reduce(
```

Comment:
instead of removing `fromEntries` helper I would put the Array.reduce logic in there

Reply:
The main problem I had with fromEntries to begin is that it feels a bit "hackish" to return an array just to flip it to an object directly after. I personally don't think keeping fromEntries() helps in this case since we'd still have to go the route of creating an array and then flipping it into an object, instead of directly going for the object. Just my 2c.

Reply:
I see your point, something like `mapValues` would be a better abstraction